lib/ansible/modules/clustering/kubevirt/kubevirt_vm.py

```diff
@@ -0,0 +1,312 @@
+            - Name of the virtual machine.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Namespace where the virtual machine exists.
+        required: true
+        type: str
+    disks:
+        description:
+            - List of dictionaries which specify disks of the virtual machine.
+            - A disk can be made accessible via four different types: I(disk), I(lun), I(cdrom), I(floppy).
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_disk
+        type: list
+    volumes:
+        description:
+            - Volume specification for the virtual machine.
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_volume
+        type: str
+    persistent:
+        description:
+            - If I(true) and C(state) is I(running) virtual machine will be created and persisted, if I(false) and C(state) running non perisistent virtual machine will be created. 
+        type: bool
+
+extends_documentation_fragment:
+  - k8s_auth_options
+  - k8s_resource_options
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+'''
+
+EXAMPLES = '''
+- name: Start virtual machine 'myvm'
+  kubevirt_vm:
+      state: running
+      name: myvm
+      namespace: vms
+
+- name: Create virtual machine 'myvm' and start it
+  kubevirt_vm:
+      state: running
+      name: myvm
+      namespace: vms
+      memory: 64M
+      disks:
+        - name: registrydisk
+          volumeName: registryvolume
+          disk:
+            bus: virtio
+
+- name: Start non-persistent virtual machine 'myvm'
+  kubevirt_vm:
+      persistent: false
+      state: running
+      name: myvm
+      namespace: vms
+      memory: 64M
+      disks:
+        - name: registrydisk
+          volumeName: registryvolume
+          disk:
+            bus: virtio
+
+- name: Remove virtual machine 'myvm'
+  kubevirt_vm:
+      state: absent
+      name: myvm 
+      namespace: vms
+'''
+
+RETURN = '''
+vm:
+    description:
+      - The virtual machine that user manages.
+      - It contains all attributes https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachine
+    returned: success
+    type: complex
+'''
+
+import copy
+import traceback
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+from openshift.helper.exceptions import KubernetesException
+
+from openshift import watch
+from openshift.dynamic.client import ResourceInstance
+from openshift.helper.exceptions import KubernetesException
+
+
+VM_ARG_SPEC = {
+    
+    'merge_type': {'type': 'list', 'choices': ['json', 'merge', 'strategic-merge']},
+    'persistent': {'type': 'bool', 'default': True},
+    'state': {
+        'type': 'str',
+        'choices': [
+            'present', 'absent', 'running', 'stopped'
+        ],
+        'default': 'present'
+    },
+    'wait': {'type': 'bool', 'default': True},
```

Comment:
Please use wait, wait_time in one of examples.

Reply:
Ok